pubsub.go

```diff
@@ -139,6 +140,10 @@ type PubSubRouter interface {
 	Leave(topic string)
 }
 
+type PubSubRouterConfig interface {
+	JoinWithProtocol(topic string, proto protocol.ID) error
```

Comment:
This seemed like the easiest way to add in topic/subscription based options. However, it's a little bizarre and I'd be fine with either replacing the interface with a simple function (non-breaking change) or investigating an interface breaking change/addition to PubSubRouter.

Reply:
We are fine we interface-breaking changes here, routers are only used internally for now.

Reply:
great, so it should be easy enough to remove this and combine it into the existing interface (or just replace/update the existing Join function)